bindgen-tests/tests/tests.rs

```diff
@@ -427,6 +428,58 @@ fn test_header_contents() {
     assert_eq!(expected, actual);
 }
 
+#[test]
+fn test_declare_safe() {
+    // prettyplease does not retain non-doc comments: https://github.com/dtolnay/prettyplease/issues/50
+
+    #[derive(Debug)]
+    struct DeclareSafe;
+
+    impl ParseCallbacks for DeclareSafe {
+        fn declare_safe(&self, item_info: ItemInfo<'_>) -> Option<String> {
+            match item_info.kind {
+                ItemKind::Function => {
+                    if item_info.name == "my_safe_func" {
+                        return Some("safe to call".to_owned());
+                    }
+                }
+                ItemKind::Var => {
+                    if item_info.name == "my_safe_var" {
+                        return Some("safe to access".to_owned());
+                    }
+                }
+                _ => todo!(),
+            }
+            None
+        }
+    }
+
+    let actual = builder()
+        .disable_header_comment()
+        .header_contents(
+            "safe.h",
+            "const int my_safe_var[3] = {1,2,3};\
+                     int my_safe_func();",
+        )
+        .formatter(Formatter::Rustfmt)
+        .parse_callbacks(Box::new(DeclareSafe))
+        .generate()
+        .unwrap()
+        .to_string();
+
+    let expected = r#"unsafe extern "C" {
+    /* Safety : "safe to access" */
+    pub safe static my_safe_var: [::std::os::raw::c_int; 3usize];
+}
+unsafe extern "C" {
+    /* Safety : "safe to call" */
+    pub safe fn my_safe_func() -> ::std::os::raw::c_int;
+}
```

Comment:
Are these comments meant to describe only the reason why they are marked as `safe`, i.e. rather than the full signature being correct? Related: https://github.com/rust-lang/rust-clippy/issues/13560.

Reply:
The specific comments are just there to test things are working, they could be replaced by `dummy reasoning for test`.

The comments can be used to express any kind of safety reasoning. Generally I'd expect people to only reason about why a function is safe to call or a const is safe to access and to rely on `bindgen` to generate correct bindings.